llama_index/node_parser/metadata_extraction.py

```diff
@@ -0,0 +1,250 @@
+"""Simple node parser."""
+from enum import Enum
+import json
+from typing import List, Optional, Sequence
+from functools import reduce
+
+from llama_index.callbacks.base import CallbackManager
+from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.constants import DEFAULT_CHUNK_OVERLAP, DEFAULT_CHUNK_SIZE
+from llama_index.langchain_helpers.text_splitter import TextSplitter, TokenTextSplitter
+from llama_index.llm_predictor.base import BaseLLMPredictor
+from llama_index.node_parser.interface import NodeParser
+from llama_index.node_parser.node_utils import get_nodes_from_document
+from llama_index.prompts.base import Prompt
+from llama_index.schema import BaseNode, Document
+
+
+class MetadataExtractionMode(str, Enum):
+    """Modes for metadata extraction.
+
+    Attributes:
+        TODO
+    """
+
+    SELF_SUMMARY = "self_summary"
+    PREV_SUMMARY = "prev_summary"
+    NEXT_SUMMARY = "next_summary"
+    PREV_NEXT_SUMMARY = "prev_next_summary"
+
+    QUESTION_EMBEDDING = "question_embedding"
+
+    SUBSECTION_LOCATION = "subsections"
+    SUBSECTION_SUMMARY_CONTEXT = "subsection_summaries"
+
+    DOCUMENT_TITLE = "doc_title"
+    DOCUMENT_KEYWORDS = "doc_keywords"
+
+
+class MetadataExtractionNodeParser(NodeParser):
```

Comment:
Perhaps better to call this something like:
1. `FeatureAugmentationNodeParser`
2. `FeatureAugmentationMetadataNodeParser`
?

Reply:
I like 2.